### ISSUE TITLE
A couple suggestions

### ISSUE THREAD

**Issue by yarrysmod** [floushee/cheat-sheets#1]
git checkout -b <new branch> <reference branch> # creates and checks out a new branch from a referenced branch (if no referenced given, then HEAD will be taken as reference)
e.g.
git checkout -b feature_JIRA_1234_short_description origin/master

git fetch -p # retrieves a copy of the remote branches and removes local copies of remote branches which have been removed already

git tag <tag_name> # generates a tag at the current HEAD position (also possible to use a -m "message here" for a short note for the created tag)
git push origin --tags # pushes all local tags to the remote repository 'origin'

**Comment by floushee** [comment]
Ahh so cool that you come up to me with information like that :)

**Comment by yarrysmod** [comment]
yeah i figured you'd like some nifty tricks, like resetting a branch to the current HEAD, like this
`git checkout -B <branch_name> HEAD`

or rebasing your current branch to that of the remote branch without a merge commit (especially helpful when working on the same branch with others)
`git pull --rebase`
